integration/samples/fail-missing-secondary-deps/secondary/package.json

```diff
@@ -0,0 +1,13 @@
+{
```

Comment:
Can we remove this test?
This is an invalid entry-point. Dependencies cannot be defined in secondary entry-points.

Reply:
I've removed this test and added one with a secondary entry point that fails at:
```
import { exampleFooBarFactory } from '@angular/core';
```

The newly added test was failing, it turns out, that also in this case the destination was not cleaned.

I've tried looking for a single place to cleanup, mainly `entryPointTransformFactory()`, but I've ended up adding a cleanup inside all the transforms provided to `entryPointTransformFactory()`:
```
    compileTs,
    writeBundles,
    writePackage,
```

Reply:
It's too much of a hassle to cleanup on build errors and not on watch. I've kept the newly added tests that are failing, but I've disabled them.

I've recreated the branch, please check the new commits (they should not be squashed upon merging).

Reply:
Resolving, the newly added disabled tests were removed, I'll create a followup PR when we also have a fix for the cleanup.